File: dynamic_listings/lua/default/developing_mechanics/jump_buffering_buffer.lua
-- ...
JUMPBUFFERTIME = 5.0
-- ...
local function update(dt)
    -- ...
    if (CONTROLS.jump.isPressed()) then
        player.hasBufferedJump = true
        player.jumpBufferCountdown = JUMPBUFFERTIME
    end
    -- Take note on how this piece is outside the "jump is pressed" section
    if (player.hasBufferedJump) then
        player.jumpBufferCountdown = player.jumpBufferCountdown - dt
    end
    if (player.on_ground) then
        if (player.jumpBufferCountdown > 0.0) then
            -- Jump
            player.jumpBufferCountdown = 0.0
            player.hasBufferedJump = false
        end
    end
    -- ...
end

File: dynamic_listings/rust/default/developing_mechanics/jump_buffering_buffer.rs
// ...
const JUMP_BUFFER_TIME: f32 = 5.;
// ...
fn update(&mut self, dt: f32) {
    //...
    if self.controls.jump.is_pressed() {
        self.player.has_buffered_jump = true;
        self.player.jump_buffer_countdown = Self::JUMP_BUFFER_TIME;
    }
    // Take note on how this piece is outside the "jump is pressed" section
    if self.player.has_buffered_jump {
        self.player.jump_buffer_countdown = self.player.jump_buffer_countdown - dt;
    }
    if self.player.on_ground {
        if self.player.jump_buffer_countdown > 0. {
            // Jump
            self.player.jump_buffer_countdown = 0.;
            self.player.has_buffered_jump = false;
        }
    }
    //...
}
